### ISSUE TITLE
IOError : Failed to fetch ads from schedd.

### ISSUE THREAD

**Issue by gourangakole** [cms-sw/genproductions#1518]
Hi Experts;

I am using CMSConnect, But get some error related to file IO I think.
Could you please comment on this.
Details log and buggy gridpack location [0] 
My cards are here [1]

[0]
https://gkole.web.cern.ch/gkole/public/MC_Contact/Madgraph5_aMCatNLO/CMSConnect/diphoton_v1/
[1]
https://github.com/gourangakole/genproductions/tree/signal_125/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/diphotonJets_5f_NLO_FXFX_Mgg80ToInf

**Comment by khurtado** [comment]
Hi Gouranga,

This seems to be a transient error related to schedd which seems to have happened on lxplus in the past [too.](https://github.com/cms-sw/genproductions/issues/1177#issuecomment-332195120) Could you try resubmitting your card?
I have enabled some extra debugging in condor to check in case it happens again.

**Comment by kdlong** [comment]
@gourangakole, did you have any luck after resubmitting?

**Comment by gourangakole** [comment]
Thanks @khurtado 
I have submitted but you know meanwhile CMSConnect switch to /local-scratch.
So again got delayed. With the new area submitted again, I do not know whether that will solve my problem or not, running for 5 days. No idea what will happen next. 

My area: /local-scratch/gkole/17Nov2017/genproductions/bin/MadGraph5_aMCatNLO

**Comment by gourangakole** [comment]
@kdlong @khurtado the bad luck or what I would say :)
Given the time after 1 week I see the failure, I am really afraid how do I finish all other N numbers of gridpack in time as a just MC contact.
Please take a look at details 
https://gkole.web.cern.ch/gkole/public/MC_Contact/Madgraph5_aMCatNLO/CMSConnect/diphoton_v2/

ESC[1;31mCommand "launch -n pilotrun" interrupted with error:
ClusterManagmentError : Some Jobs are in a Hold/... state. Error messages are below.Please try to investigate or contact the IT team. 
        ClusterId 902866 with HoldReason: Error from glidein_54_224051249@hd-38-28.dice.priv: Proxy about to expire
Please report this bug on https://bugs.launchpad.net/mg5amcnlo
More information is found in '/local-scratch/gkole/17Nov2017/genproductions/bin/MadGraph5_aMCatNLO/diphotonJets_5f_NLO_FXFX_Mgg80ToInf/diphotonJets_5f_NLO_FXFX_Mgg80ToInf_gridpack/work/processtmp/pilotrun_tag_1_debug.log'.
Please attach this file to your report.ESC[0m

**Comment by gourangakole** [comment]
It would be really helpful if anyone just submit jobs in "cmsconnect" ....then I would be more confident that there is nothing worng in the card.
becuase, I do not see any such subprocess in the error file.

**Comment by khurtado** [comment]
@gourangakole
Looks like your grid proxy needed renewal and the system stopped accepting more jobs because of that. The run card itself didn't fail this time.
There is a proxy watcher process that send email reminders, but yours might have been removed by another gridpack that finished before, so it didn't send you anymore reminders. This is related to #1556

Could you renew your proxy now? You seem to have another gridpack running.

**Comment by gourangakole** [comment]
@khurtado 
Let me summarize:
I have created the new area.
Created proxy and submitted two jobs:
Now the currently I see "only" one job:

condor_q gkole -af:j ClusterId Owner Cmd Args
897635.0 897635 gkole /usr/local/wrapper/connect_wrapper.sh ajob1 13 F 0 0

Though I see my one masterjobs still in the backgroud:
[gkole@login MadGraph5_aMCatNLO]$ ps aux | grep gkole | grep cmsconnect
gkole    1164888  0.0  0.0 106316  2936 ?        S    Nov17   0:00 /bin/bash ./submit_cmsconnect_gridpack_generation.sh TTGG_0Jets_5f_NLO cards/production/2017/13TeV/TTGG_0Jets_5f_NLO 1 4 Gb
gkole    2467532  0.0  0.0 103388  2020 pts/3    S+   07:15   0:00 grep cmsconnect

So I do not understand why is that? 
I still see my proxy is valid:
[gkole@login MadGraph5_aMCatNLO]$ voms-proxy-info
subject   : /C=TW/O=AP/OU=GRID/CN=Gouranga Kole 105330/CN=2754907921
issuer    : /C=TW/O=AP/OU=GRID/CN=Gouranga Kole 105330
identity  : /C=TW/O=AP/OU=GRID/CN=Gouranga Kole 105330
type      : RFC compliant proxy
strength  : 1024 bits
path      : /tmp/x509up_u100191
timeleft  : 47:36:48

what is the minimum "timeleft" required to run a job?

**Comment by khurtado** [comment]
@gourangakole:
So, some sites start holding jobs if proxies are about to expire  in X hours. This can be e.g 8 hours, or less than 24 hours usually. 
The watcher starts reminding users when their proxies are 48 hours or less to expire. 
You job ran at `T2_UK_SGrid_Bristol`. It seems they hold jobs if the user's proxy is less than 48 hours. This is new to me.
 I think I will have to change the reminder to remind people starting 72 hours.

**Comment by gourangakole** [comment]
@khurtado Ok. Now renewed the proxy to 192 hrs. 
Sumitted new jobs again "diphotonJets_5f_NLO_FXFX_Mgg80ToInf" (NOW)

Now my other query I do not understand: 
if 
[gkole@login MadGraph5_aMCatNLO]$ ps aux | grep gkole | grep cmsconnect
gkole    1164888  0.0  0.0 106316  2936 ?        S    Nov17   0:00 /bin/bash ./submit_cmsconnect_gridpack_generation.sh TTGG_0Jets_5f_NLO cards/production/2017/13TeV/TTGG_0Jets_5f_NLO 1 4 Gb
gkole    2644025  0.0  0.0 103388  2080 pts/3    S+   07:51   0:00 grep cmsconnect

why "myq" gives one job
[gkole@login MadGraph5_aMCatNLO]$ myq
897635.0 897635 gkole /usr/local/wrapper/connect_wrapper.sh ajob1 13 F 0 0

does the other jobs for the master one "1164888" will resume? 
and 
have to

**Comment by khurtado** [comment]
So, I just changed the reminder to notify when timeleft < 72 hours.

Given your gridpack takes quite long and has a few jobs running for days, I would suggest the following before you start a new trial:

1. Change [this line](https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/submit_cmsconnect_gridpack_generation.sh#L133):
 to 
```
export CONDOR_SET_MAXWALLTIMES="500,2820"
```
That way, short jobs will match fast. Long jobs will match slower but it will guarantee +48 hours of running time (I have seen your last jobs can take about that long) . I think those numbers fit this workflow better and should be done a bit faster overall. 

2. Comment [this line](https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/submit_cmsconnect_gridpack_generation.sh#L278) for now, until #1556 is fixed.

Make sure you always renew your proxy on time. You can see if the watcher is running by doing
```
condor_q gkole -constraint 'JobUniverse == 12'
```
I just started a watcher for you.

3. Update to the latest master version of genproductions and OPTIONALLY, you can apply this PR #1557. This fixes bugs in madgraph in the cluster manager, but it hasn't been thoroughly tested yet. It basically makes resubmission work when there was a cvmfs issue.

**Comment by khurtado** [comment]
@gourangakole 
Regarding your other gridpack:
So, `ps` is showing you the command you are still running:
```
gkole 1164888 0.0 0.0 106316 2936 ? S Nov17 0:00 /bin/bash ./submit_cmsconnect_gridpack_generation.sh TTGG_0Jets_5f_NLO cards/production/2017/13TeV/TTGG_0Jets_5f_NLO 1 4 Gb
```

The logfile associated to your production [1], shows the master is just waiting for 1 job to finish. so everything is going normally there, after that job is done the master will decide what to do (submit more jobs, etc):

```
...
INFO:  Idle: 0,  Running: 1,  Completed: 54 [  212h 49m  ]
INFO:  Idle: 0,  Running: 1,  Completed: 54 [  212h 54m  ]
```

[1] `
/local-scratch/gkole/17Nov2017/genproductions/bin/MadGraph5_aMCatNLO/TTGG_0Jets_5f_NLO.log`

**Comment by gourangakole** [comment]
@khurtado After your suggestion again back to the square one :(
details can be found here:

https://gkole.web.cern.ch/gkole/public/MC_Contact/Madgraph5_aMCatNLO/CMSConnect/diphoton_v3/

error:
^[[1;31mCommand "launch -n pilotrun" interrupted with error:
ClusterManagmentError : Some Jobs are in a Hold/... state. Error messages are below.Please try to investigate or contact the IT team\
.
        ClusterId 1158232 with HoldReason: Error from glidein_54_782021788@hd-38-22.dice.priv: Proxy about to expire
Please report this bug on https://bugs.launchpad.net/mg5amcnlo
More information is found in '/local-scratch/gkole/special_test_diPhoton/genproductions/bin/MadGraph5_aMCatNLO/diphotonJets_5f_NLO_F\
XFX_Mgg80ToInf/diphotonJets_5f_NLO_FXFX_Mgg80ToInf_gridpack/work/processtmp/pilotrun_tag_1_debug.log'.
Please attach this file to your report.^[[0m
quit

Just to tell you I have proxy:
timeleft  : 139:22:14

**Comment by khurtado** [comment]
@gourangakole Wow, this seems like problem with `T2_UK_SGrid_Bristol` configuration. You had the same issue with the same site before.
I have blacklisted it from the default list of sites for now and will report the issue to them. 

Please, go ahead and resubmit again, change [this line](https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/submit_cmsconnect_gridpack_generation.sh#L125) to:
```
  export CONDOR_RELEASE_HOLDCODES="26:119,13,30:256,12:28,4:0"
```
this will make held jobs due to proxy expiration be released automatically by madgraph. You should hopefully be able to finish your run this time. Sorry about the issue.

**Comment by gourangakole** [comment]
@khurtado 
again back to the initial point where I started.
Failed the gridpack just running after 18 days even after all your suggestions. 
^[[1;31mCommand "launch -n pilotrun" interrupted with error:
IOError : Failed to fetch ads from schedd.

https://gkole.web.cern.ch/gkole/public/MC_Contact/Madgraph5_aMCatNLO/CMSConnect/diphoton_v4/

**Comment by khurtado** [comment]
@gourangakole : I'm sorry to read that. 
Looking into your history, the main problem is the fact that you had at least a couple of jobs needing longer than 48 hours to run, so they were running but the remote site glideins timed out many times. There was one running for almost 60 hours before this issue came up.
The Global Pool has a limit of 48 hours per glidein, so jobs usually can't run longer than that, hence the "2820" max requested value. There is a very few sites allowing more than that, you were running on some of them, but I'm not sure how many hours your jobs will need to successfully complete. 

I know CERN HLT allows weeks, but I'm not sure if they allow analysis jobs or only production jobs, so I will run some tests to match jobs with longer than 70 or 100 hours and let you know.

That said, there is an upcoming OSG maintenance scheduled from Dec 25 until Jan 1st, so this gridpack in particular might not be able to finish before that.

You might want to try to get this particular gridpack done at e.g the LPC CAF for example, since they don't have such walltime restriction per job.
If that is not an option and since (despite the odds) resubmitting doesn't hurt, you could try updating genproductions code and resubmit using:

```export CONDOR_SET_MAXWALLTIMES="500,4200,6000"```

You have only 1 job requiring longer than 15 hours, about 10 jobs requiring near 48 hours and at least 2 requiring longer than 50, there is another 2 which we don't know how long will they really need, but hopefully 70 hours will be enough, those jobs will take quite a while to match but they should eventually do so. The 6000 minutes might not even match, depending what job types CERN HLT accept, but that's the last resort, I will find out about it.

**Comment by khurtado** [comment]
@gourangakole While looking into your logfile, I realized you apparently did not change the values in this line for your gridpack (I can see jobs requesting 960 and 2160 mins when 500 was not enough):

https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/submit_cmsconnect_gridpack_generation.sh#L133

I think that's the reason your gridpack took 18 days before failing. Most of your jobs require less than 8 hours to complete, but about 10-15 of them require +40 hours, so it doesn't make sense to request less than that in your case, so please make sure you use

```
export CONDOR_SET_MAXWALLTIMES="500,4200,6000"
```
if you try again.

**Comment by gourangakole** [comment]
@khurtado 
If I look back https://github.com/cms-sw/genproductions/issues/1518#issuecomment-347011472
there you have suggested to use 
export CONDOR_SET_MAXWALLTIMES="500,4200,6000"

then I have continued to use that.
it is really tricky. if able to get lpc account I shall try there.

**Comment by gourangakole** [comment]
Hi @khurtado ..I have tried to run in LPC as well.
But  it fails 3 times .. I have added the full log [*]       
 file missing: /uscms_data/d3/gkole/MC_contact/genproductions/bin/MadGraph5_aMCatNLO/diphotonJets_5f_NLO_FXFX_Mgg80ToInf/diphotonJets_5f_NLO_FXFX_Mgg80ToInf_\
gridpack/work/processtmp/SubProcesses/P2_gd_aadg/GF5/log_MINT0.txt
            Fails 3 times
            No resubmition.  ^[[0m
INFO: All jobs finished
INFO:  Idle: 0,  Running: 0,  Completed: 589 [  192h 7m  ]
Can anyone try please? I do not know where is the problem. 
[*]https://gkole.web.cern.ch/gkole/public/MC_Contact/Madgraph5_aMCatNLO/LPC/diphoton_v1/

**Comment by khurtado** [comment]
@gourangakole You can try asking in the hypernews. There are people there using the LPC with gridpacks, so they might have a clue.

**Comment by gourangakole** [comment]
Yes Luca ..this can be closed since I have opened another issue #1640  closely related to this. 
Thank you.
